Mathlib/Tactic/ToAdditive.lean

```diff
@@ -305,16 +309,21 @@ def applyReplacementFun (e : Expr) : MetaM Expr := do
   let isRelevant : Name → ℕ → Bool := fun nm i ↦ i == (relevantArgAttr.find? env nm).getD 0
   return aux
       (findTranslation? <| ← getEnv) reorderFn (ignoreArgsAttr.find? env)
-      (fixedNumeralAttr.find? env) isRelevant e
+      (fixedNumeralAttr.find? env) isRelevant (← getBoolOption `trace.to_additive_detail) e
 where /-- Implementation of `applyReplacementFun`. -/
   aux (findTranslation? : Name → Option Name)
     (reorderFn : Name → List ℕ) (ignore : Name → Option (List ℕ))
-    (fixedNumeral : Name → Option Bool) (isRelevant : Name → ℕ → Bool) : Expr → Expr :=
+    (fixedNumeral : Name → Option Bool) (isRelevant : Name → ℕ → Bool) (trace : Bool) :
+    Expr → Expr :=
   Lean.Expr.replaceRec fun r e ↦ Id.run do
+    if trace then
+      dbg_trace s!"replacing at {e}"
```

Comment:
I don't think it's a good idea to use `dbg_trace` for logging.  It doesn't use MessageData, and the output is not interleaved with the other trace messages (`dbg_trace` just prints things to stdout).

We should either put this function in `CoreM` or leave out the logging.

Reply:
Putting this function in `CoreM` would require generalizing `memoFix` to monads, so that it memoizes the output value of a monad application. I don't know if that is doable, but I don't know how to do that.

I find  the `dbg_trace` useful when debugging, and they will only ever fire with `trace.to_additive_detail` set to true, which almost nobody ever does (and never happens on the master branch of mathlib). That said, if you think this bad, I'll remove the `dbg_trace` (I could also put them behind a new option `to_additive_debug`).

Reply:
Ah, I didn't realize it was only set with `to_additive_detail`.  I still don't think it's a good solution, but at least it's contained.